### PR TITLE
(refactor) replace magic URL with BITFINEX_REST_URL constant

### PR DESCRIPTION
**Pull request by 0xferit**
Substituted 

`async with client.get("https://api-pub.bitfinex.com/v2/conf/pub:list:pair:exchange", timeout=10) as response:` 

with 

`async with client.get(f"{BITFINEX_REST_URL}/conf/pub:list:pair:exchange", timeout=10) as response:`

to eliminate [magic URL](https://en.wikipedia.org/wiki/Magic_number_(programming)).

The rest of the changes were introduced by isort commit hook.

----

P.S. I periodically rebase and force-push because the branch is getting stale due to updates in `hummingbot/development`.

### CODE REVIEW

**Comment by nikspz** [comment]
@0xferit 
Currently on this PR `Unexpected error occurred when listening to order book streams` still happened, however was able to place order/ got some orders fills 


Crosschecked with development, was able to place orders/ get order fills there as well.


[logs_bitfinex1.log](https://github.com/hummingbot/hummingbot/files/13737058/logs_bitfinex1.log)

```
2023-12-21 06:30:09,344 - 2965466 - hummingbot.connector.exchange.bitfinex.bitfinex_api_order_book_data_source - ERROR - Unexpected error occurred when listening to order book streams. Retrying in 5 seconds...
Traceback (most recent call last):
  File "/home/nikita/bitfinex6718/hummingbot/core/data_type/order_book_tracker_data_source.py", line 78, in listen_for_subscriptions
    ws: WSAssistant = await self._connected_websocket_assistant()
  File "/home/nikita/bitfinex6718/hummingbot/core/data_type/order_book_tracker_data_source.py", line 200, in _connected_websocket_assistant
    raise NotImplementedError
NotImplementedError
2023-12-21 06:30:10,083 - 2965466 - hummingbot.connector.exchange.bitfinex.bitfinex_exchange - INFO - Order filled -100 out of -100.00000 of the limit sell order sell-DOGE-USDT-1703140194002030
2023-12-21 06:30:10,084 - 2965466 - hummingbot.strategy.pure_market_making.pure_market_making - INFO - (DOGE-USDT) Maker sell order of 100 DOGE filled. [clock=2023-12-21 06:30:10+00:00]
2023-12-21 06:30:10,111 - 2965466 - hummingbot.core.event.event_reporter - EVENT_LOG - {"timestamp": 1703140210.0, "order_id": "sell-DOGE-USDT-1703140194002030", "trading_pair": "DOGE-USDT", "trade_type": "TradeType.SELL", "order_type": "OrderType.LIMIT", "price": "0.091079999999999994297894545525196008384227752685546875", "amount": "100", "trade_fee": {"percent": "0", "percent_token": null, "flat_fees": [{"token": "USDT", "amount": "-0.009108"}]}, "exchange_trade_id": "1478236134", "exchange_order_id": "", "leverage": 1, "position": "NIL", "event_name": "OrderFilledEvent", "event_source": "bitfinex"}
2023-12-21 06:30:10,111 - 2965466 - hummingbot.connector.exchange.bitfinex.bitfinex_exchange - INFO - The market sell order sell-DOGE-USDT-1703140194002030 has completed according to Bitfinex user stream.
2023-12-21 06:30:10,112 - 2965466 - hummingbot.strategy.pure_market_making.pure_market_making - INFO - (DOGE-USDT) Maker sell order sell-DOGE-USDT-1703140194002030 (100.00000 DOGE @ 0.09108 USDT) has been completely filled. [clock=2023-12-21 06:30:10+00:00]
2023-12-21 06:30:10,124 - 2965466 - hummingbot.core.event.event_reporter - EVENT_LOG - {"timestamp": 1703140210.0, "order_id": "sell-DOGE-USDT-1703140194002030", "base_asset": "DOGE", "quote_asset": "USDT", "base_asset_amount": "100", "quote_asset_amount": "9.10800", "order_type": "OrderType.LIMIT", "exchange_order_id": null, "event_name": "SellOrderCompletedEvent", "event_source": "bitfinex"}
2023-12-21 06:30:10,345 - 2965466 - hummingbot.connector.exchange.bitfinex.bitfinex_api_order_book_data_source - ERROR - Unexpected error occurred when listening to order book streams. Retrying in 5 seconds...
Traceback (most recent call last):
  File "/home/nikita/bitfinex6718/hummingbot/core/data_type/order_book_tracker_data_source.py", line 78, in listen_for_subscriptions
    ws: WSAssistant = await self._connected_websocket_assistant()
  File "/home/nikita/bitfinex6718/hummingbot/core/data_type/order_book_tracker_data_source.py", line 200, in _connected_websocket_assistant
    raise NotImplementedError
NotImplementedError
2023-12-21 06:30:11,346 - 2965466 - hummingbot.connector.exchange.bitfinex.bitfinex_api_order_book_data_source - ERROR - Unexpected error occurred when listening to order book streams. Retrying in 5 seconds...
Traceback (most recent call last):
  File "/home/nikita/bitfinex6718/hummingbot/core/data_type/order_book_tracker_data_source.py", line 78, in listen_for_subscriptions
    ws: WSAssistant = await self._connected_websocket_assistant()
  File "/home/nikita/bitfinex6718/hummingbot/core/data_type/order_book_tracker_data_source.py", line 200, in _connected_websocket_assistant
    raise NotImplementedError
NotImplementedError
2023-12-21 06:30:12,349 - 2965466 - hummingbot.connector.exchange.bitfinex.bitfinex_api_order_book_data_source - ERROR - Unexpected error occurred when listening to order book streams. Retrying in 5 seconds...
Traceback (most recent call last):
  File "/home/nikita/bitfinex6718/hummingbot/core/data_type/order_book_tracker_data_source.py", line 78, in listen_for_subscriptions
    ws: WSAssistant = await self._connected_websocket_assistant()
  File "/home/nikita/bitfinex6718/hummingbot/core/data_type/order_book_tracker_data_source.py", line 200, in _connected_websocket_assistant
    raise NotImplementedError
NotImplementedError
2023-12-21 06:30:13,351 - 2965466 - hummingbot.connector.exchange.bitfinex.bitfinex_api_order_book_data_source - ERROR - Unexpected error occurred when listening to order book streams. Retrying in 5 seconds...
Traceback (most recent call last):
  File "/home/nikita/bitfinex6718/hummingbot/core/data_type/order_book_tracker_data_source.py", line 78, in listen_for_subscriptions
    ws: WSAssistant = await self._connected_websocket_assistant()
  File "/home/nikita/bitfinex6718/hummingbot/core/data_type/order_book_tracker_data_source.py", line 200, in _connected_websocket_assistant
    raise NotImplementedError
NotImplementedError
2023-12-21 06:30:14,001 - 2965466 - hummingbot.strategy.pure_market_making.pure_market_making - INFO - (DOGE-USDT) Canceling the limit order buy-DOGE-USDT-1703140194001665. [clock=2023-12-21 06:30:14+00:00]
2023-12-21 06:30:14,353 - 2965466 - hummingbot.connector.exchange.bitfinex.bitfinex_api_order_book_data_source - ERROR - Unexpected error occurred when listening to order book streams. Retrying in 5 seconds...
Traceback (most recent call last):
  File "/home/nikita/bitfinex6718/hummingbot/core/data_type/order_book_tracker_data_source.py", line 78, in listen_for_subscriptions
    ws: WSAssistant = await self._connected_websocket_assistant()
  File "/home/nikita/bitfinex6718/hummingbot/core/data_type/order_book_tracker_data_source.py", line 200, in _connected_websocket_assistant
    raise NotImplementedError
NotImplementedError
2023-12-21 06:30:15,006 - 2965466 - hummingbot.connector.exchange.bitfinex.bitfinex_exchange - INFO - Successfully canceled order buy-DOGE-USDT-1703140194001665.
2023-12-21 06:30:15,021 - 2965466 - hummingbot.core.event.event_reporter - EVENT_LOG - {"timestamp": 1703140215.0, "order_id": "buy-DOGE-USDT-1703140194001665", "exchange_order_id": null, "event_name": "OrderCancelledEvent", "event_source": "bitfinex"}
2023-12-21 06:30:15,355 - 2965466 - hummingbot.connector.exchange.bitfinex.bitfinex_api_order_book_data_source - ERROR - Unexpected error occurred when listening to order book streams. Retrying in 5 seconds...
Traceback (most recent call last):
  File "/home/nikita/bitfinex6718/hummingbot/core/data_type/order_book_tracker_data_source.py", line 78, in listen_for_subscriptions
    ws: WSAssistant = await self._connected_websocket_assistant()
  File "/home/nikita/bitfinex6718/hummingbot/core/data_type/order_book_tracker_data_source.py", line 200, in _connected_websocket_assistant
    raise NotImplementedError
NotImplementedError
2023-12-21 06:30:15,817 - 2965466 - hummingbot.connector.exchange.bitfinex.bitfinex_exchange - INFO - The market order buy-DOGE-USDT-1703140194001665 has failed/been canceled according to order status API.
2023-12-21 06:30:15,826 - 2965466 - hummingbot.core.event.event_reporter - EVENT_LOG - {"timestamp": 1703140215.0, "order_id": "buy-DOGE-USDT-1703140194001665", "exchange_order_id": null, "event_name": "OrderCancelledEvent", "event_source": "bitfinex"}
2023-12-21 06:30:16,381 - 2965466 - hummingbot.connector.exchange.bitfinex.bitfinex_api_order_book_data_source - ERROR - Unexpected error occurred when listening to order book streams. Retrying in 5 seconds...
Traceback (most recent call last):
  File "/home/nikita/bitfinex6718/hummingbot/core/data_type/order_book_tracker_data_source.py", line 78, in listen_for_subscriptions
    ws: WSAssistant = await self._connected_websocket_assistant()
  File "/home/nikita/bitfinex6718/hummingbot/core/data_type/order_book_tracker_data_source.py", line 200, in _connected_websocket_assistant
    raise NotImplementedError
NotImplementedError

```

**Comment by nikspz** [comment]
Did this PR fixed https://github.com/hummingbot/hummingbot/issues/5919 for you?

**Comment by 0xferit** [comment]
I think you misunderstood, @nikspz. This PR does not intend to fix #5919 or is related to #6590, but it is a simple refactoring to fix a code quality issue I stumbled upon.

I'm indeed working on #6590 and #5919, but I'm still trying to understand the codebase. And when I stumble upon a problem, I fix it or at least submit an issue to let others fix it.

File: hummingbot/connector/exchange/bitfinex/bitfinex_api_order_book_data_source.py [hunk 1]
```diff
@@ -87,7 +71,7 @@ def __init__(self, trading_pairs: Optional[List[str]] = None):
     async def fetch_trading_pairs() -> List[str]:
         try:
             async with aiohttp.ClientSession() as client:
-                async with client.get("https://api-pub.bitfinex.com/v2/conf/pub:list:pair:exchange", timeout=10) as response:
+                async with client.get(f"{BITFINEX_REST_URL}/conf/pub:list:pair:exchange", timeout=10) as response:
```
Comment (0xferit) [inline hummingbot/connector/exchange/bitfinex/bitfinex_api_order_book_data_source.py:74]: The only change. The rest of the changes were introduced by the `isort`Git commit hook.

**Review by cardosofede — APPROVED** [review]
LGTM!
This is more a style/remove hardcode value improvement so we should just merge it in

**Pull request merged**